community/detectors/flowise_exposed_ui/src/main/java/com/google/tsunami/plugins/detectors/exposedui/flowise/FlowiseExposedUiDetector.java

```diff
@@ -0,0 +1,143 @@
+
+  @Inject
+  FlowiseExposedUiDetector(@UtcClock Clock utcClock, HttpClient httpClient) {
+    this.utcClock = checkNotNull(utcClock);
+    this.httpClient = checkNotNull(httpClient).modify().setFollowRedirects(false).build();
+  }
+
+  @Override
+  public DetectionReportList detect(
+      TargetInfo targetInfo, ImmutableList<NetworkService> matchedServices) {
+    logger.atInfo().log("FlowiseExposedUiDetector starts detecting.");
+
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(NetworkServiceUtils::isWebService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    String targetUri = NetworkServiceUtils.buildWebApplicationRootUrl(networkService);
+    String targetApiUri = targetUri + "api/v1/apikey";
+
+    HttpResponse response;
+    try {
+      // plain GET request to check Flowise UI availability.
+      response =
+          httpClient.send(HttpRequest.get(targetUri).withEmptyHeaders().build(), networkService);
+      if (!(response.bodyString().isPresent() && response.bodyString().get().contains("Flowise"))) {
+        return false;
+      }
+
+      // Main request that performs vulnerability check.
+      response =
+          httpClient.send(
+              HttpRequest.get(targetApiUri)
+                  .setHeaders(HttpHeaders.builder().addHeader("x-request-from", "internal").build())
+                  .build(),
+              networkService);
+      return response.status().code() != 401;
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log("Unable to query Flowise.");
+      return false;
+    }
+  }
+
+  private DetectionReport buildDetectionReport(
+      TargetInfo targetInfo, NetworkService vulnerableNetworkService) {
+    return DetectionReport.newBuilder()
+        .setTargetInfo(targetInfo)
+        .setNetworkService(vulnerableNetworkService)
+        .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+        .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+        .setVulnerability(
+            Vulnerability.newBuilder()
+                .setMainId(
+                    VulnerabilityId.newBuilder()
+                        .setPublisher("TSUNAMI_COMMUNITY")
+                        .setValue("FLOWISE_UI_EXPOSED"))
+                .setSeverity(Severity.HIGH)
+                .setTitle("Flowise UI Exposed")
+                .setDescription("Flowise UI instance is exposed without proper authentication.")
+                .setRecommendation(
+                    "Secure the Flowise UI by implementing proper authentication.\n"
+                        + "Consider restricting access to trusted networks only.")
```

Comment:
Could you provide a more detailed recommendation about how to configure authentication in Flowise?